src/functions-and-arrays.js

```diff
@@ -1,41 +1,100 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
 
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  }
+  else if (num1 < num2) {
+    return num2;
+  }
+  else {
+    return num1 || num2;
+  }  
+}
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(arr){
+  let longestWord = "";
+    if (arr.length === 0) return null;// if (!arr.length) 
 
+    for(let i=0; i < arr.length; i++) {
+      if (arr[i].length > longestWord.length) {
+        longestWord = arr[i];
+      };
+    }
+    return longestWord;
+}
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
+function sumNumbers(numbers) {
+  let sum = 0;
+  numbers.forEach((number)=>{
+    sum = sum + number;
+  })
+  return sum;
+}
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
+function sum(mixedValues) {
+  
+  let sum = 0; 
+  for(let i=0; i<mixedValues.length; i++) {
+    if (typeof mixedValues[i] === "string") {
+      mixedValues[i] = mixedValues[i].length;
+    }   
+    if (typeof mixedValues[i] === "object") {
+      throw new Error;
+    }   
+    sum+= mixedValues[i];
+  }  
+  return sum;
+}
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function sumNumbers(numbers) {
+  let sum = 0; 
```

Comment:
you already have the function `sumNumbers()` declared in line 35, you do not have to do it again here, it is the same file